surefire-providers/surefire-junit4/src/main/java/org/apache/maven/surefire/junit4/JUnit4Provider.java

```diff
@@ -270,18 +276,57 @@ private void executeTestSet( Class<?> clazz, RunListener reporter, Notifier noti
         }
     }
 
+    private static int indexOf( List<ResolvedTest> tests, Description description )
```

Comment:
If you do something like this, you are on wrong way.
The problem is that you wanted to get to the end within one day, and that's the problem.
You should follow those 5 steps, you would not get to the end after you have implemented first 4 steps but would would do 85% of your work without removing anything.
If you are not sure in the beginning, it still right way to write an integration test because it is always needed and it is the vision you want to emphase to everyone.

Reply:
Thanks, I will close this PR and send another one. I've also written up a JIRA issue explaining a bit more about the motivation: https://issues.apache.org/jira/browse/SUREFIRE-2041

Reply:
Please see https://github.com/apache/maven-surefire/pull/495 for my take II on this issue.